src/gui/torrentcontenttreeview.cpp

```diff
@@ -106,6 +108,29 @@ void TorrentContentTreeView::keyPressEvent(QKeyEvent *event)
     }
 }
 
+void TorrentContentTreeView::mouseDoubleClickEvent(QMouseEvent *event)
+{
+    const QPersistentModelIndex index = indexAt(event->pos());
+    if (index.isValid())
+    {
+        const QVariant checkValue = index.data(Qt::CheckStateRole);
+        if (checkValue.isValid() && !checkValue.isNull())
+        {
+            QStyleOptionViewItem option;
+            option.initFrom(this);
+            option.rect = visualRect(index);
+            const QRect checkRect = style()->subElementRect(QStyle::SE_CheckBoxClickRect, &option, this);
+            if (!checkRect.isValid() || !checkRect.contains(event->pos()))
+                emit itemDoubleClicked(index);
+        }
+        else
+        {
+            emit itemDoubleClicked(index);
+        }
+    }
+    return QTreeView::mouseDoubleClickEvent(event);
```

Comment:
One last comment, you should drop the `return`.

Reply:
I personally prefer return since it's a "natural" end of a function, I will remove it if you insist

Reply:
The `return` in such function doesn't fit in our coding style, it is not used anywhere else.

Reply:
Personally I like return void in case it has some value but this style was rejected. Anyway it is meaningless in the function end.